rpm/__init__.py

```diff
@@ -71,6 +71,18 @@ def get_sitepackages_and_suffixes(interpreter):
     for interpreter in interpreters:
         if not Path(interpreter).is_file():
             continue
+        # skip interpreters with a different major.minor version than the
+        # running interpreter
+        # importing their rpm module has most likely catastrophic consequences
+        i_major, i_minor, _ = (
+            subprocess.check_output([interpreter, "--version"])
+            .decode()
+            .strip()
+            .removeprefix("Python ")
+            .split(".")
+        )
+        if i_major != majorver or i_minor != minorver:
```

Comment:
```suggestion
        if (i_major, i_minor) != (majorver, minorver):
```

I'd probably prefer this, since it's a bit easier to read.